rotkehlchen/chain/evm/decoding/cctp/decoder.py

```diff
@@ -0,0 +1,145 @@
+            msg_aggregator=msg_aggregator,
+        )
+        self.token_messenger = token_messenger
+        self.message_transmitter = message_transmitter
+        self.asset_identifier = asset_identifier
+
+    def _decode_deposit(self, context: DecoderContext) -> DecodingOutput:
+        if not self.base.is_tracked(user_address := hex_or_bytes_to_address(context.tx_log.topics[3])):  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        to_chain = hex_or_bytes_to_int(context.tx_log.data[64:96])
+        deposit_amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[0:32]),
+            token_decimals=USDC_DECIMALS,
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset.identifier == self.asset_identifier and
+                event.balance.amount == deposit_amount and
+                event.location_label == user_address
+            ):
+                try:
+                    chain_info = f' from {self.evm_inquirer.chain_id.label()} to {CCTP_DOMAIN_MAPPING[to_chain].label()}'  # noqa: E501
+                except KeyError:
+                    log.error(f'Could not find chain ID {to_chain} for CCTP transfer from {self.evm_inquirer.chain_name}')  # noqa: E501
+                    chain_info = ''
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = f'Bridge {event.balance.amount} USDC{chain_info} via CCTP'
+                event.counterparty = CPT_CCTP
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_withdraw(self, context: DecoderContext) -> DecodingOutput:
+        if not self.base.is_tracked(user_address := hex_or_bytes_to_address(context.tx_log.topics[1])):  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        deposit_amount = token_normalized_value_decimals(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[0:32]),
+            token_decimals=USDC_DECIMALS,
+        )
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset.identifier == self.asset_identifier and
+                event.balance.amount == deposit_amount and
+                event.location_label == user_address
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = f'Bridge {event.balance.amount} USDC via CCTP'
+                event.counterparty = CPT_CCTP
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_message_received(self, context: DecoderContext) -> DecodingOutput:
```

Comment:
If this is what you decode just to get the chain info during withdrawal can you add this to a short docstring description?